gio-subclass/src/auto/action_map.rs

```diff
@@ -0,0 +1,144 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ e895709+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 47c69e6)
+// DO NOT EDIT
+
+#[allow(unused_imports)]
+use libc::{c_int, c_char, c_uchar, c_float, c_uint, c_double,
+    c_short, c_ushort, c_long, c_ulong,
+    c_void, size_t, ssize_t, intptr_t, uintptr_t, time_t, FILE};
+
+#[allow(unused_imports)]
+use glib_ffi::{gboolean, gconstpointer, gpointer, GType};
+
+use gio;
+use gio_ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+use gobject_subclass::anyimpl::*;
+use gobject_subclass::object::*;
+
+
+pub trait ActionMapImpl: AnyImpl + 'static {
+
+    fn add_action(&self, action_map: &gio::ActionMap, action: &gio::Action);
+
+    fn lookup_action(&self, action_map: &gio::ActionMap, action_name: &str) -> Option<gio::Action>;
+
+    fn remove_action(&self, action_map: &gio::ActionMap, action_name: &str);
+
+}
+
+any_impl!(ActionMapImpl);
+
+pub trait ActionMapImplStatic<T: ObjectType>: 'static {
+    fn get_impl<'a>(&self, imp: &'a T::ImplType) -> &'a ActionMapImpl;
+}
+
+struct ActionMapStatic<T: ObjectType>{
+    imp_static: *const ActionMapImplStatic<T>
+}
+
+
+unsafe extern "C" fn action_map_add_action<T: ObjectType>
+(gptr: *mut gio_ffi::GActionMap, action: *mut gio_ffi::GAction)
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_map_get_type())
+                                     as *const ActionMapStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+    imp.add_action(&wrap, &from_glib_none(action));
```

Comment:
Why not `&from_glib_borrow(action)`?

Reply:
Not sure, I'll look into that. Either this is a bug in the gir definitions or in the generator.